src/dsp_tools/commands/ingest_xmlupload/apply_ingest.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import annotations
+
+import glob
+from typing import cast
+
+import pandas as pd
+from lxml import etree
+
+from dsp_tools.commands.ingest_xmlupload.user_information import IngestInformation
+from dsp_tools.models.exceptions import InputError
+
+
+def get_mapping_dict_from_file() -> dict[str, str]:
+    """
+    This functions returns the information to replace the original filepaths with the identifier from dsp-ingest.
+
+    Returns:
+        dictionary with original: identifier from dsp-ingest
+
+    Raises:
+        InputError: if no file was found
+    """
+    if filepath := glob.glob("mapping-*.csv"):
+        df = pd.read_csv(filepath[0])
+        print(f"The file '{filepath[0]}' is used to map the internal SIPI image IDs to the original filepaths.")
+        return dict(zip(df["original"].tolist(), df["derivative"].tolist()))
+    else:
+        raise InputError("No mapping csv file was found in the directory.")
+
+
+def replace_filepath_with_sipi_uuid(
+    xml_tree: etree._ElementTree[etree._Element],
+    orig_path_2_uuid_filename: dict[str, str],
+) -> tuple[etree._ElementTree[etree._Element], IngestInformation]:
```

Comment:
an alternative to changing the XML (which I always feel a bit uneasy when we do that) would be to loop over the created objects and modify them according to the mapping. For now, with the `XMLResource` object being tightly coupled to XML which it requires to create an instance, this would probably not be super convenient - but in the long run, once we have decoupled these things, I would actually prefer that approach.

Reply:
That sounds like a good suggestion.

Reply:
What I do not want to loose though is the check if all the media in the mapping file was used or not, and which media  was not found. At the end of the substitution rather than stopping as soon as one was unsuccessful. And I am not sure how to implement that without having to hand around a lot of information or doing the same work (checking for bitstream) twice, which I am also not a fan of...